src/btree/bt_handle.c

```diff
@@ -110,6 +110,12 @@ __wt_btree_open(WT_SESSION_IMPL *session, const char *op_cfg[])
     /* Initialize and configure the WT_BTREE structure. */
     WT_ERR(__btree_conf(session, &ckpt));
 
+    /* Set special flags for the history store table. */
+    if (strcmp(dhandle->name, WT_HS_URI) == 0) {
+        F_SET(btree, WT_BTREE_HS);
+        F_SET(btree, WT_BTREE_NO_LOGGING);
+    }
```

Comment:
I'd move this into `__btree_conf()`, that's where the `WT_BTREE` structure generally gets set up. Most of it is `cfg` based, but some of it (at the end), is more general. If you have a reason for it being here, no objection, of course.

Reply:
Moved into `__btree_conf()` function.